Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/Shadow/DirectionalLightShadowCalculator.azsli

```diff
@@ -0,0 +1,416 @@
+
+    const uint cascadeCount = ViewSrg::m_directionalLightShadows[m_lightIndex].m_cascadeCount;
+    const float3 shadowOffset = ComputeNormalShadowOffset(ViewSrg::m_directionalLightShadows[m_lightIndex].m_normalShadowBias, m_worldNormal, ViewSrg::m_directionalLightShadows[m_lightIndex].m_shadowmapSize);
+
+    for (uint index = 0; index < cascadeCount; ++index)
+    {        
+        float4 lightSpacePos = mul(worldToLightViewMatrices[index], float4(m_worldPos + shadowOffset, 1.));
+        lightSpacePos.z += shadowBias;
+        
+        const float4 clipSpacePos = mul(lightViewToShadowmapMatrices[index], lightSpacePos);                
+        m_shadowCoords[index] = clipSpacePos.xyz / clipSpacePos.w;
+    }
+            
+    if (m_receiverShadowPlaneBiasEnable)
+    {
+        [unroll]
+        for(int i = 0 ; i < ViewSrg::MaxCascadeCount ; ++i)
+        {
+            m_shadowPosDX[i] = ddx_fine(m_shadowCoords[i]);
+            m_shadowPosDY[i] = ddy_fine(m_shadowCoords[i]);        
+        }    
+    }        
+}
+
+bool2 DirectionalShadowCalculator::IsShadowed(const float3 shadowCoord, const uint indexOfCascade)
+{
+    static const float PixelMargin = 1.5; // avoiding artifact between cascade levels.
+    static const float DepthMargin = 1e-8; // avoiding artifact when near depth bounds.
+
+    // size is the shadowap's width and height.
+    const uint size = ViewSrg::m_directionalLightShadows[m_lightIndex].m_shadowmapSize;
+
+    const uint cascadeCount = ViewSrg::m_directionalLightShadows[m_lightIndex].m_cascadeCount;
+
+    [branch]
+    if (shadowCoord.x >= 0. && shadowCoord.x * size < size - PixelMargin &&
+        shadowCoord.y >= 0. && shadowCoord.y * size < size - PixelMargin)
+    {
+        const float3 coord = float3(shadowCoord.xy, indexOfCascade);
+        const float depthInShadowmap = m_directionalLightShadowmap.Sample(PassSrg::LinearSampler, coord).r;
+        // Test the case where the shadowed pixel is nearer than the maximal depth
+        // of the cascade.  Otherwise, escalate the next cascade.
+        if (shadowCoord.z < 1. - DepthMargin)
+        {
+            const float depthDiff = shadowCoord.z - depthInShadowmap;
+            return bool2(true, (depthDiff > m_slopeBias[indexOfCascade]));
+        }
+    }
+    return bool2(false, false);
+}
+
+float DirectionalShadowCalculator::GetVisibility()
+{   
+    ComputeShadowCoords();
+
+    // N.L early out
+    const float3 lightDirection = normalize(SceneSrg::m_directionalLights[m_lightIndex].m_direction);
+    float NdotL = dot(m_worldNormal, -lightDirection);    
+
+    // Todo: slope bias exhibits noticeable artifacts, especially with sharp normal values derived from fullscreen
+    // depth values which accentuate geometry polygons. Revisit the math and determine if this can be remedied or
+    // whether it should be removed altogether. In the meantime, normal shadow bias is plenty capable.
+    static const bool useSlopeBias = false;
```

Comment:
My memory of applying this technique is that the slope-scaled bias is applied when rendering the shadow map itself (with a known camera pose), and not something applied when sampling the shadow map, mainly since this can be accomplished by changing the rasterizer state and leveraging the hardware. I don't think it's uncommon to rely on a slope bias in the rasterizer state, and subsequently do a view-based bias after also.